src/main/java/bisq/desktop/main/dao/proposal/ProposalListItem.java

```diff
@@ -82,11 +103,83 @@ public ProposalListItem(Proposal proposal,
         txConfidenceIndicator.setPrefSize(24, 24);
         txConfidenceIndicator.setTooltip(tooltip);
 
+        actionButton = new AutoTooltipButton();
+        actionButton.setMinWidth(70);
+        actionButtonIconView = new ImageView();
+
         chainHeightListener = (observable, oldValue, newValue) -> setupConfidence();
         bsqWalletService.getChainHeightProperty().addListener(chainHeightListener);
         setupConfidence();
 
         bsqBlockChainChangeDispatcher.addBsqBlockChainListener(this);
+
+        phaseChangeListener = (observable, oldValue, newValue) -> {
+            applyState(newValue, proposal.getVoteResult());
+        };
+
+        voteResultChangeListener = (observable, oldValue, newValue) -> {
+            applyState(daoPeriodService.getPhaseProperty().get(), newValue);
+        };
+
+        daoPeriodService.getPhaseProperty().addListener(phaseChangeListener);
+        proposal.getVoteResultProperty().addListener(voteResultChangeListener);
+        applyState(daoPeriodService.getPhaseProperty().get(), proposal.getVoteResult());
+    }
+
+    private void applyState(DaoPeriodService.Phase newValue, VoteResult voteResult) {
+        actionButton.setText("");
+        actionButton.setVisible(false);
+        actionButton.setOnAction(null);
+
+        switch (newValue) {
+            case UNDEFINED:
+                break;
+            case COMPENSATION_REQUESTS:
```

Comment:
Perhaps we should rename this PROPOSALS?

Reply:
Yes, good point. Will do for next PR.